---a/net/minecraft/entity/ai/EntityLookHelper.java
+++b/net/minecraft/entity/ai/EntityLookHelper.java
@@ -5,8 +5,10 @@
 import net.minecraft.entity.EntityLivingBase;
 import net.minecraft.util.MathHelper;
 
-public class EntityLookHelper
-{
+import org.bukkit.craftbukkit.TrigMath;
+
+public class EntityLookHelper {
+
     private EntityLiving entity;
     private float deltaLookYaw;
     private float deltaLookPitch;
@@ -16,17 +16,14 @@
     private double posZ;
     private static final String __OBFID = "CL_00001572";
 
-    public EntityLookHelper(EntityLiving p_i1613_1_)
-    {
+    public EntityLookHelper(EntityLiving p_i1613_1_) {
         this.entity = p_i1613_1_;
     }
 
-    public void setLookPositionWithEntity(Entity p_75651_1_, float p_75651_2_, float p_75651_3_)
-    {
+    public void setLookPositionWithEntity(Entity p_75651_1_, float p_75651_2_, float p_75651_3_) {
         this.posX = p_75651_1_.posX;
 
-        if (p_75651_1_ instanceof EntityLivingBase)
-        {
-            this.posY = p_75651_1_.posY + (double)p_75651_1_.getEyeHeight();
-        }
-        else
-        {
+        if (p_75651_1_ instanceof EntityLivingBase) {
+            this.posY = p_75651_1_.posY + (double) p_75651_1_.getEyeHeight();
+        } else {
             this.posY = (p_75651_1_.getEntityBoundingBox().minY + p_75651_1_.getEntityBoundingBox().maxY) / 2.0D;
         }
 
@@ -40,8 +40,7 @@
         this.isLooking = true;
     }
 
-    public void setLookPosition(double p_75650_1_, double p_75650_3_, double p_75650_5_, float p_75650_7_, float p_75650_8_)
-    {
+    public void setLookPosition(double p_75650_1_, double p_75650_3_, double p_75650_5_, float p_75650_7_, float p_75650_8_) {
         this.posX = p_75650_1_;
         this.posY = p_75650_3_;
         this.posZ = p_75650_5_;
@@ -50,78 +50,65 @@
         this.isLooking = true;
     }
 
-    public void onUpdateLook()
-    {
+    public void onUpdateLook() {
         this.entity.rotationPitch = 0.0F;
 
-        if (this.isLooking)
-        {
+        if (this.isLooking) {
             this.isLooking = false;
             double d0 = this.posX - this.entity.posX;
-            double d1 = this.posY - (this.entity.posY + (double)this.entity.getEyeHeight());
+            double d1 = this.posY - (this.entity.posY + (double) this.entity.getEyeHeight());
             double d2 = this.posZ - this.entity.posZ;
-            double d3 = (double)MathHelper.sqrt_double(d0 * d0 + d2 * d2);
-            float f = (float)(Math.atan2(d2, d0) * 180.0D / Math.PI) - 90.0F;
-            float f1 = (float)(-(Math.atan2(d1, d3) * 180.0D / Math.PI));
+            double d3 = (double) MathHelper.sqrt_double(d0 * d0 + d2 * d2);
+            // CraftBukkit start - Math -> TrigMath
+            float f = (float) (TrigMath.atan2(d2, d0) * 180.0D / 3.1415927410125732D) - 90.0F;
+            float f1 = (float) (-(TrigMath.atan2(d1, d3) * 180.0D / 3.1415927410125732D));
+            // CraftBukkit end
             this.entity.rotationPitch = this.updateRotation(this.entity.rotationPitch, f1, this.deltaLookPitch);
             this.entity.rotationYawHead = this.updateRotation(this.entity.rotationYawHead, f, this.deltaLookYaw);
-        }
-        else
-        {
+        } else {
             this.entity.rotationYawHead = this.updateRotation(this.entity.rotationYawHead, this.entity.renderYawOffset, 10.0F);
         }
 
         float f2 = MathHelper.wrapAngleTo180_float(this.entity.rotationYawHead - this.entity.renderYawOffset);
 
-        if (!this.entity.getNavigator().noPath())
-        {
-            if (f2 < -75.0F)
-            {
+        if (!this.entity.getNavigator().noPath()) {
+            if (f2 < -75.0F) {
                 this.entity.rotationYawHead = this.entity.renderYawOffset - 75.0F;
             }
 
-            if (f2 > 75.0F)
-            {
+            if (f2 > 75.0F) {
                 this.entity.rotationYawHead = this.entity.renderYawOffset + 75.0F;
             }
         }
     }
 
-    private float updateRotation(float p_75652_1_, float p_75652_2_, float p_75652_3_)
-    {
+    private float updateRotation(float p_75652_1_, float p_75652_2_, float p_75652_3_) {
         float f3 = MathHelper.wrapAngleTo180_float(p_75652_2_ - p_75652_1_);
 
-        if (f3 > p_75652_3_)
-        {
+        if (f3 > p_75652_3_) {
             f3 = p_75652_3_;
         }
 
-        if (f3 < -p_75652_3_)
-        {
+        if (f3 < -p_75652_3_) {
             f3 = -p_75652_3_;
         }
 
         return p_75652_1_ + f3;
     }
 
-    public boolean func_180424_b()
-    {
+    public boolean func_180424_b() {
         return this.isLooking;
     }
 
-    public double func_180423_e()
-    {
+    public double func_180423_e() {
         return this.posX;
     }
 
-    public double func_180422_f()
-    {
+    public double func_180422_f() {
         return this.posY;
     }
 
-    public double func_180421_g()
-    {
+    public double func_180421_g() {
         return this.posZ;
     }
-}
+}
